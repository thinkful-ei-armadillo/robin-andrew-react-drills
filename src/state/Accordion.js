import React, { Component } from "react";

export default class Accordion extends Component {
  static defaultProps = {
    sections: []
  };
  state = {
    currentClick: null
  };

  renderButtons(sections) {
    const newVar = sections.map((el, index) => {
      return (
        <li key={index} id={index}>
          <button onClick={() => this.handleButtonClick(index)}>
            {el.title}
          </button>
          {this.state.currentClick === index && <p>{el.content}</p>}
        </li>
      );
    });

    return newVar;
  }

  handleButtonClick = index => {
    this.setState({ currentClick: index });
  };

  render() {
    const { sections } = this.props;
    console.log({ sections });
    return <ul>{this.renderButtons(sections)}</ul>;
  }
}
