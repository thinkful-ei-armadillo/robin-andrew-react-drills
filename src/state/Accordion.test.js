import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Accordion from "./Accordion";
import sections from "./STORE";

describe("", () => {
  it("renders without error", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Accordion sections={sections} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders the UI as expected", () => {
    const tree = renderer.create(<Accordion sections={sections} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
