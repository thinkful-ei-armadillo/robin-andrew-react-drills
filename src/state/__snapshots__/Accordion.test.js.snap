// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<ul>
  <li
    id={0}
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    id={1}
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    id={2}
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
